src/libcore/slice.rs

```diff
@@ -897,6 +911,24 @@ impl<'a, T> Iter<'a, T> {
     ///
     /// This has the same lifetime as the original slice, and so the
     /// iterator can continue to be used while this exists.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// // First, we declare a type which implements `Iter` trait (&[usize here]):
+    /// let slice = &[1, 2, 3];
+    ///
+    /// // Then, we get the iterator:
+    /// let mut iter = slice.iter();
```

Comment:
it'd be nice if there was a bit more whitespace here, this feels cramped.

```
// comment
code

// comment
code

// comment
code
```

what do you think?